examples/canvas_more/canvas_more/app.py

```diff
@@ -0,0 +1,216 @@
+from contextlib import contextmanager
+from math import pi, cos, sin, atan
+
+import toga
+from toga.style import Pack
+from toga.style.pack import COLUMN
+from travertino.size import at_least
+
+
+class ExampleCanvasApp(toga.App):
+    def startup(self):
+        # Initial width and height for the Canvas
+        iwidth = 600
+        iheight = 400
+
+        # Set up main window
+        self.main_window = toga.MainWindow(title=self.name)
+
+        # Setup the style selection box
+        self._style_select = toga.Selection(
+            items=list(self.styles), style=Pack(height=32), on_select=self.set_style
+        )
+        # Setup the canvas
+        self._canvas = toga.Canvas(style=Pack(flex=1), on_resize=self.resize_contents)
+        self._canvas.intrinsic = \
+            Pack.IntrinsicSize(width=at_least(iwidth), height=at_least(iheight))
+
+        # Add the content on the main window
+        box = toga.Box(
+            style=Pack(direction=COLUMN),
+            children=[self._style_select, self._canvas]
+        )
+        self.main_window.content = box
+
+        # Setup canvas transformation objects use to handle window moving and
+        # resizing so that actual drawing code can use a fixed range of
+        # coordinates
+        self._canvas_translate = self._canvas.translate(iwidth/2, iheight/2)
+        self._canvas_scale = self._canvas.scale(iwidth/6, iheight/6)
+        self.render_drawing()
+
+        # Show the main window
+        self.main_window.show()
+
+    def render_drawing(self):
+        """Renders the dawing on the canvas by calling a set of internal
+        functions that draw the various shapes
+        """
+        def draw_rects():
+            for ctx_idx, step in enumerate(range(5, 0, -1)):
+                with self.draw_context(ctx_idx) as ctx:
+                    p = step/6.
+                    ctx.rect(-p, -p, 2*p, 2*p)
+
+        def draw_arcs():
+            for ctx_idx, step in enumerate(range(5, 0, -1)):
+                with self.draw_context(ctx_idx, 0, 0) as ctx:
+                    radius = .25+step/8.
+                    angle = pi*(.5+ctx_idx/8.)
```

Comment:
This is completely legal Python, but stylistic preference is to always in include the 0 in floats (i.e., `0.25` not `.25`, and `8.0` not `8.`).

Reply:
Ok will fix soon